app/src/main/java/org/fossasia/openevent/app/core/event/list/EventsListAdapter.java

```diff
@@ -28,26 +29,38 @@
 class EventsListAdapter extends RecyclerView.Adapter<EventsListAdapter.EventRecyclerViewHolder>
     implements StickyRecyclerHeadersAdapter<HeaderViewHolder>, Filterable {
 
-    private final List<Event> events;
+    private List<Event> events;
     private final List<Event> selectedEvents = new ArrayList<>();
 
     private final Bus bus;
-    private final EventsPresenter eventsPresenter;
     private boolean sortByName;
 
     private boolean isLongPressed;
 
-    EventsListAdapter(List<Event> events, Bus bus, EventsPresenter eventsPresenter) {
+    private final EventListFragment fragment;
+
+    EventsListAdapter(List<Event> events, Bus bus, EventListFragment fragment) {
         this.events = events;
         this.bus = bus;
-        this.eventsPresenter = eventsPresenter;
+        this.fragment = fragment;
     }
 
-    public void categorizeEvents() {
-
+    public void setEvents(List<Event> events) {
```

Comment:
Can't quite figure out the use of this method. Where are you using it ? You can simply use `updateList` only.